### PR TITLE
Integration of exam approval (for practice tests/guests)

### PR DESCRIPTION
**Pull request by burzum619**
This PR includes code in TDS_Student for calling the new ExamService approve accommodations endpoint. This endpoint will be called when an guest user needs approval for additional/selectable exam accommodations.

### CODE REVIEW

File: student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    examRepository.approveAccommodations(oppInstance.getExamId(), request);
+  }
+  
+  @Override
+  public List<Accommodations> getApproved(OpportunityInstance opportunityInstance, String testKey, boolean isGuestSession) throws ReturnStatusException {
+    return legacyAccommodationsService.getApproved(opportunityInstance, testKey, isGuestSession);
+  }
+  
+  
+  private Map<Integer, Set<String>> parseSegmentAccommodationStrings(List<String> segmentsAccommodationData) {
```
Comment (jtreuting) [inline student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java:80]: It would be nice to get some unit tests around this logic to make sure it handles edge cases okay

File: student.library/src/main/java/tds/student/sql/abstractions/ExamRepository.java [hunk 2]
```diff
@@ -11,4 +14,5 @@
  */
 public interface ExamRepository {
   Response<Exam> openExam(final OpenExamRequest openExamRequest) throws ReturnStatusException;
+  void approveAccommodations(final UUID examId, final ApproveAccommodationsRequest approveAccommodationsRequest) throws ReturnStatusException;
```
Comment (millerts) [inline student.library/src/main/java/tds/student/sql/abstractions/ExamRepository.java]: Can we add JavaDocs to the new method?  I created one for the open exam on the feature branch

File: student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  @Override
+  public void approve(OpportunityInstance oppInstance, List<String> segmentsAccommodationData) throws ReturnStatusException {
+
+    if (isLegacyCallsEnabled) {
+      legacyAccommodationsService.approve(oppInstance, segmentsAccommodationData);
+    }
+
+    if (!isRemoteExamCallsEnabled || segmentsAccommodationData == null) {
```
Comment (millerts) [inline student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java]: Why are we bailing from the remote call if the segmentsAccommodationData is null even though we call the legacy code?  It seems that the segmentsAccommodation being null should short circuit it all?

If not, can we add comments as to why it is like this?

File: student/src/main/java/tds/student/sql/repository/RemoteExamRepository.java [hunk 4]
```diff
@@ -47,4 +56,25 @@ public RemoteExamRepository(@Qualifier("integrationRestTemplate") final RestTemp
...
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    HttpEntity<?> requestHttpEntity = new HttpEntity<>(approveAccommodationsRequest, headers);
+
+    UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(String.format("%s/%s/accommodations", examUrl, examId));
+
+    try {
+      restTemplate.exchange(
+        builder.build().encode().toUri(),
```
Comment (millerts) [inline student/src/main/java/tds/student/sql/repository/RemoteExamRepository.java]: The encode is unnecessary

File: student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java (hunk 3 above)
Comment (burzum619) [reply student/src/main/java/tds/student/services/remote/RemoteAccommodationsService.java]: The legacy code also bails out early if segmentData == null;  (line 275,275 in AccommodationService.java) Ultimately I guess we can just bail out immediately if segmentData == null, and not call either service. Strange that it doesnt result in any sort of warning or error.
